### PR TITLE
DOC: Fix `bounds` parameter documentation

### LINKED ISSUE #7492: Inconsistent documentation of `bounds` parameter in modeling classes
Currently to set `bounds` parameter in `Lorentz1D` class takes a dictionary with parameter name as key and min, max as _both_ tuple and list. But in docs, it is not mentioned clearly. The detail is also different in its base class. I have put the sections below:

screen shot 2018-05-23 at 23 23 13
**Description in `Lorentz1D1 doc page**

screen shot 2018-05-23 at 23 23 58
**Desctription in base class**

As @astrofrog suggested, it should be either `{parameter_name: (min, max)}` or `{parameter_name: value}`

@pllim suggested adding example(s) for the same in the documentation.

I would like to work on this.

**Note:** I used the parameter and it supports both tuple and list as value.

### PR DESCRIPTION
**Pull request by swapsha96**
Resolves #7492. 

Todo: Add example.

### CODE REVIEW

File: astropy/stats/info_theory.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def bayesian_info_criterion_lsq(ssr, n_params, n_samples):
     >>> y = 3 * np.exp(-0.5 * (x - 1.3)**2 / 0.8**2)
     >>> y += np.random.normal(0., 0.2, x.shape)
     >>> # Fit the data using a Box model
-    >>> t_init = models.Trapezoid1D(amplitude=1., x_0=0., width=1., slope=0.5)
+    >>> t_init = models.Trapezoid1D(amplitude=1., x_0=0., width=1., slope=0.5, bounds=["x_0": (-1., 1.)])
```
Comment (pllim) [inline astropy/stats/info_theory.py]: `bounds` should be a dictionary with curly brackets. Same as the example below. This is causing `doctest` failures.

Comment (swapsha96) [reply astropy/stats/info_theory.py]: @pllim I made the changes accordingly.

**Comment by pllim** [comment]
Looks like your addition to `stats` example broke `doctest` as the function now gives different result:
```python
_______ [doctest] astropy.stats.info_theory.bayesian_info_criterion_lsq ________
169     >>> g_init = models.Gaussian1D(amplitude=1., mean=0, stddev=1.)
170     >>> fit_g = fitting.LevMarLSQFitter()
171     >>> g = fit_g(g_init, x, y)
172     >>> # Compute the mean squared errors
173     >>> ssr_t = np.sum((t(x) - y)*(t(x) - y))
174     >>> ssr_g = np.sum((g(x) - y)*(g(x) - y))
175     >>> # Compute the bics
176     >>> bic_t = bayesian_info_criterion_lsq(ssr_t, 4, x.shape[0])
177     >>> bic_g = bayesian_info_criterion_lsq(ssr_g, 3, x.shape[0])
178     >>> bic_t - bic_g # doctest: +FLOAT_CMP
Expected:
    30.644474706065466
Got:
    212.83039990507456
```

File: astropy/stats/info_theory.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def bayesian_info_criterion_lsq(ssr, n_params, n_samples):
     >>> bic_t = bayesian_info_criterion_lsq(ssr_t, 4, x.shape[0])
     >>> bic_g = bayesian_info_criterion_lsq(ssr_g, 3, x.shape[0])
     >>> bic_t - bic_g # doctest: +FLOAT_CMP
-    30.644474706065466
+    212.83039990507456
```
Comment (pllim) [inline astropy/stats/info_theory.py]: Is it okay to just change the value like that? Maybe someone wants to check if the answer is indeed correct? @crawfordsm ?

**Comment by swapsha96** [comment]
@pllim @crawfordsm I ran the example and then changed the value to make sure what I did is right. I hope this helps:
screen shot 2018-05-26 at 00 28 43

**Comment by swapsha96** [comment]
@pllim Now it failed like this:
screen shot 2018-05-26 at 01 14 31

**Comment by swapsha96** [comment]
@pllim now it failed like this:
screen shot 2018-05-26 at 01 14 31

**Comment by pllim** [comment]
Hmm 0.005 difference seems too big to be floating point uncertainty. Is there a reason why bounds need to be added to that stats example?

**Comment by swapsha96** [comment]
@pllim Actually, I am not sure if it's important to use it here. It helps to set range of variation of a parameter during fitting a model. Is there any other good place to put an example?

**Comment by swapsha96** [comment]
@pllim I fixed the problem. The problem that I created myself. I set the range from -1 to +1 but after *correct* fitting, the parameter value is around `1.27` and I didn't let that happen. Not I set the range from -5 to +5 and it works! My bad. I guess it's good to merge. 😄

**Pull request closed without merging**